dali/base/dadfs.cpp

```diff
@@ -4474,7 +4474,12 @@ protected: friend class CDistributedFilePart;
         if (isPathSepChar(newPath.charAt(newPath.length()-1)))
             newPath.setLength(newPath.length()-1);
         newPath.remove(0, myBase.length());
-        newdir.append(baseDir).append(newPath);
+        // Ensure one path separator between baseDir and newPath
+        newdir.append(addPathSepChar(baseDir, psc));
```

Comment:
This is potentially altering 'baseDir' (which is used later on line 4499).
It would be better not to alter baseDir (unless there's a reason for it? in which case there should be a comment explaining why). Assuming only a side-effect, better to change to:
```
newdir.append(baseDir);
addPathSepChar(newdir, psc));
```

Reply:
Good point I'll make that change.

Reply:
The side effect turned out to be necessary- `baseDir` is used in that loop to build the path used for the rename operation. The `newdir` variable was used to set file directory metadata. I thought this approach using a function was more clear, but if you feel it is overkill I can revert to the previous approach and comment instead.